### ISSUE TITLE
Expire and remove expired subscriptions

### ISSUE THREAD

**Issue by jelmd** [mattermost/mattermost-server#14316]
There are a lot of users, which have an typo in their e-mail address when subscribing to a team. Since they never get the related "verify" e-mail, the related username is blocked forever without any reason. Note that this can therefore be used as an DoS vector as well!

Mattermost should record, when it sends out a verification e-mail. When the user verifies itself, the expiration date gets removed.  Otherwise, when it expires it should trigger a garbage collection, where the record gets removed - and thus the username is free again. An admin should have a Channel or Team setting to set the expiration time. At least in a university context it would probably 1h, rarely more than 1 day.

**Comment by amyblais** [comment]
I'll ask our team to verify if this is a bug or an improvement request,

**Comment by amyblais** [comment]
@jelmd What Mattermost server version and OS are you on?

**Comment by jelmd** [comment]
@amyblais version 5.3.0 and started testing 5.21.0 both on Ubuntu 18.04 (bionic).

**Comment by amyblais** [comment]
Have you seen this issue also on v5.21?

**Comment by jelmd** [comment]
Yes, i.e. at least I cannot find any setting for this. If there is a builtin setting, how long should I wait for checking?

**Comment by amyblais** [comment]
This is a feature request but I opened a ticket here: https://mattermost.atlassian.net/browse/MM-24978.

**Comment by lieut-data** [comment]
Thanks for reporting this issue and for the community discussion around it. After multiple years of inactivity, we're closing this to help keep our issue tracker focused on active work. If this is still relevant, please feel free to open a new issue with updated context and we'll be happy to take another look.
